core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -0,0 +1,492 @@
+              ? InventoryMenuUtils.smallMenu(match, verboseStatsTitle, items)
+              : InventoryMenuUtils.progressiveMenu(match, verboseStatsTitle, items);
+    }
+  }
+
+  InventoryMenu endOfMatchMenu;
+
+  private ItemStack createToolItem(MatchPlayer player) {
+    ItemStack tool = new ItemStack(Material.RED_SANDSTONE);
+    ItemMeta meta = tool.getItemMeta();
+    Component displayName =
+        TranslatableComponent.of("match.stats.title", TextColor.GREEN, TextDecoration.BOLD);
+    Component lore = TranslatableComponent.of("setting.lore", TextColor.GRAY);
+    meta.setDisplayName(TextTranslations.translateLegacy(displayName, player.getBukkit()));
+    meta.setLore(Lists.newArrayList(TextTranslations.translateLegacy(lore, player.getBukkit())));
+    meta.addItemFlags(ItemFlag.values());
+    tool.setItemMeta(meta);
+    return tool;
+  }
+
+  @EventHandler
+  public void onToolClick(PlayerInteractEvent event) {
+    if (!verboseStats) return;
+    Action action = event.getAction();
+    if ((action == Action.RIGHT_CLICK_AIR || action == Action.RIGHT_CLICK_BLOCK)
+        && match.isFinished()) {
+      ItemStack item = event.getPlayer().getItemInHand();
+
+      if (item.getType() == Material.RED_SANDSTONE) {
+        MatchPlayer player = match.getPlayer(event.getPlayer());
+        if (player == null) return;
+        endOfMatchMenu.display(player);
+      }
+    }
+  }
+
+  private Map.Entry<UUID, Integer> sortStats(Map<UUID, Integer> map) {
+    return map.entrySet().stream().max(Comparator.comparingInt(Map.Entry::getValue)).orElse(null);
+  }
+
+  private Map.Entry<UUID, Double> sortStatsDouble(Map<UUID, Double> map) {
+    return map.entrySet().stream()
+        .max(Comparator.comparingDouble(Map.Entry::getValue))
+        .orElse(null);
+  }
+
+  Component getMessage(
+      String messageKey, Map.Entry<UUID, ? extends Number> mapEntry, TextColor color) {
+    return TranslatableComponent.of(
+        messageKey,
+        playerName(mapEntry.getKey()),
+        numberComponent(String.valueOf(mapEntry.getValue()), color));
+  }
+
+  public static Component numberComponent(int stat, TextColor color) {
+    return TextComponent.of(stat, color, TextDecoration.BOLD);
+  }
+
+  public static Component numberComponent(String stat, TextColor color) {
+    return TextComponent.of(stat, color, TextDecoration.BOLD);
+  }
+
+  @EventHandler
+  public void onPlayerLeave(PlayerQuitEvent event) {
+    Player player = event.getPlayer();
+    if (allPlayerStats.containsKey(player.getUniqueId()))
+      cachedUsernames.put(player.getUniqueId(), player.getName());
+  }
+
+  @EventHandler
+  public void onPlayerJoin(PlayerJoinEvent event) {
+    UUID playerUUID = event.getPlayer().getUniqueId();
+    cachedUsernames.remove(playerUUID);
+  }
+
+  private Component playerName(UUID playerUUID) {
+    return PlayerComponent.of(
+        Bukkit.getPlayer(playerUUID),
+        cachedUsernames.getOrDefault(playerUUID, "Unknown"),
```

Comment:
There's a translatable for unknown, it's used in the tablist for unresolved authors, use it here